src/orion/executor/single_backend.py

```diff
@@ -17,13 +52,30 @@ class SingleExecutor(BaseExecutor):
     The submitted functions are wrapped with ``functools.partial``
     which are then executed in ``wait()``.
 
+    Notes
+    -----
+    The tasks are started when wait is called
+
     """
 
     def __init__(self, n_workers=1, **config):
         super(SingleExecutor, self).__init__(n_workers=1)
 
     def wait(self, futures):
-        return [future() for future in futures]
+        return [future.get() for future in futures]
+
+    def async_get(self, futures, timeout=0.01):
+        if len(futures) == 0:
+            return []
+
+        results = []
+        try:
```

Comment:
No loop because we don't expect to have more than one future at a time?

Reply:
We could still have multiple futures using async_get(). :thinking:

Reply:
We cant have multiple futures with async_get() because async_get() is what "schedules" the jobs (for the single_backend)
So no work has started until `fut.get()` is called and we know that none of the other futures are ready because we have not called `.get` yet